google/cloud/pubsub/internal/extend_leases_with_retry_test.cc

```diff
@@ -112,56 +130,148 @@ TEST(ExtendLeasesWithRetry, FailurePermanentError) {
+                             Property(&ModifyAckDeadlineRequest::ack_ids,
+                                      ElementsAre("test-001", "test-002"))))
+          .WillOnce(Return(ByMove(make_ready_future(MakeStatusWithDetails({
+              {"test-001", "TRANSIENT_FAILURE_1"},
+              {"test-002", "TRANSIENT_FAILURE_2"},
+          })))));
+    }
   }
+  auto mock_batch_callback =
+      std::make_shared<pubsub_testing::MockBatchCallback>();
 
   google::cloud::testing_util::ScopedLog log;
   ModifyAckDeadlineRequest request;
   request.add_ack_ids("test-001");
   request.add_ack_ids("test-002");
   request.add_ack_ids("test-003");
-  auto result = ExtendLeasesWithRetry(mock, CompletionQueue(mock_cq), request);
+  auto result =
+      ExtendLeasesWithRetry(mock, CompletionQueue(mock_cq), request,
+                            mock_batch_callback, /*enable_otel=*/false);
   EXPECT_THAT(result.get(), StatusIs(StatusCode::kUnknown));
   auto const log_lines = log.ExtractLines();
   EXPECT_THAT(log_lines, Contains(HasSubstr("ack_id=test-001")));
   EXPECT_THAT(log_lines, Contains(HasSubstr("ack_id=test-002")));
   EXPECT_THAT(log_lines, Contains(HasSubstr("ack_id=test-003")));
 }
 
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+TEST(ExtendLeasesWithRetry, OtelSuccess) {
+  auto mock = std::make_shared<MockSubscriberStub>();
+
+  auto mock_cq = std::make_shared<MockCompletionQueueImpl>();
+
+  {
+    ::testing::InSequence sequence;
+    EXPECT_CALL(*mock, AsyncModifyAckDeadline(
+                           _, _, _,
+                           Property(&ModifyAckDeadlineRequest::ack_ids,
+                                    ElementsAre("test-001", "test-002"))))
+        .WillOnce(Return(ByMove(make_ready_future(MakeTransient()))));
+    EXPECT_CALL(*mock_cq, MakeRelativeTimer)
+        .WillOnce(Return(ByMove(make_ready_future(
+            make_status_or(std::chrono::system_clock::now())))));
+    EXPECT_CALL(*mock, AsyncModifyAckDeadline(
+                           _, _, _,
+                           Property(&ModifyAckDeadlineRequest::ack_ids,
+                                    ElementsAre("test-001", "test-002"))))
+        .WillOnce(Return(ByMove(make_ready_future(Status{}))));
+  }
+  auto mock_batch_callback =
+      std::make_shared<pubsub_testing::MockBatchCallback>();
+  EXPECT_CALL(*mock_batch_callback, StartModackSpan).Times(2);
+  EXPECT_CALL(*mock_batch_callback, EndModackSpan).Times(2);
+  EXPECT_CALL(*mock_batch_callback, ModackEnd("test-001")).Times(2);
+  EXPECT_CALL(*mock_batch_callback, ModackEnd("test-002")).Times(2);
+
+  ModifyAckDeadlineRequest request;
+  request.add_ack_ids("test-001");
+  request.add_ack_ids("test-002");
+
+  auto result =
+      ExtendLeasesWithRetry(mock, CompletionQueue(mock_cq), request,
+                            mock_batch_callback, /*enable_otel=*/true);
+
+  EXPECT_STATUS_OK(result.get());
+}
+
+TEST(ExtendLeasesWithRetry, SuccessWithOtelEnabled) {
```

Comment:
I couldn't see how this test was different (apart from some whitespace differences) to the previous one.

Reply:
You're right I think they are the same. I think it was left over from a merge. Removed.